### ISSUE TITLE
Support for React <18

### ISSUE THREAD

**Issue by Atmos4** [rspack-contrib/storybook-rsbuild#34]
Right now, there is a problem when trying to use a version that is lower than 18.

```
Failed to compile, check the errors for troubleshooting.
File: javascript/esm|C:\Users\Atmos4\Workspace\rsbuild-react-storybook\node_modules\@storybook\react-dom-shim\dist\react-18.mjs:1:1
  × Resolve error: Can't resolve 'react-dom/client' in 'C:\Users\Atmos4\Workspace\rsbuild-react-storybook\node_modules\@storybook\react-dom-shim\dist'
   ╭─[1:1]
 1 │ import React, { useRef, useLayoutEffect } from 'react';
 2 │ import ReactDOM from 'react-dom/client';
   ·                      ──────────────────
 3 │
 4 │ var nodes=new Map,WithCallback=({callback,children})=>{let once=useRef();return useLayoutEffect(()=>{once.current!==callback&&(once.current=callback,callback());},[callback]),children},renderElement=async(node,el,rootOptions)=>{let root=await getReactRoot(el,rootOptions);return new Promise(resolve=>{root.render(React.createElement(WithCallback,{callback:()=>resolve(null)},node));})},unmountElement=(el,shouldUseNewRootApi)=>{let root=nodes.get(el);root&&(root.unmount(),nodes.delete(el));},getReactRoot=async(el,rootOptions)=>{let root=nodes.get(el);return root||(root=ReactDOM.createRoot(el,rootOptions),nodes.set(el,root)),root};
```

I believe that if used correctly, `@storybook/react-dom-shim` will target the correct React DOM import (`react-dom` instead of `react-dom/client`).

## Reproduction

From the `/sandboxes/react-rsbuild`, change the React version to 16 or 17.

**Comment by fi3ework** [comment]
It should work. Could you try the sandbox demo in a new repository as multi-version React in a monorepo is a mess.

**Comment by Atmos4** [comment]
@fi3ework here is [the replicate of the repo](https://codesandbox.io/p/github/Atmos4/rsbuild-storybook-demo/main?import=true) I used for testing. Unfortunately I couldn't get the sandbox to run because storybook gets grumpy, but you should be able to access my code!

Btw will you publish a new npm version for storybook 8.2? It's annoying that I cannot test it out with the latest.

**Comment by fi3ework** [comment]
I've located where's wrong. Fixing tha.

**Comment by fi3ework** [comment]
Fixed in 0.0.7.

**Comment by Atmos4** [comment]
Just tested it, it works like a charm! I will update the repo I posted in the thread.

Thank you for your help! ❤️
